### PR TITLE
Update gem2rpm templates

### PR DESCRIPTION
**Pull request by ekohl**
This aligns all templates on the modern standard, which allows using gemspec_{add,remove}_dep macros without modification.

### CODE REVIEW

**Comment by ekohl** [comment]
There are more gem2rpm fixes needed. I'll collect them here.

File: gem2rpm/foreman_plugin.spec.erb [hunk 1]
```diff
@@ -101,15 +107,11 @@ Documentation for %{name}.
...
-
-%setup -q -D -T -n  %{gem_name}-%{version}
-
-gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
+%setup -q -n  %{gem_name}-%{version}
 
 %build
 # Create the gem as gem install only works on a gem file
-gem build %{gem_name}.gemspec
+gem build ../%{gem_name}-%{version}.gemspec
```
Comment (ehelms) [inline gem2rpm/foreman_plugin.spec.erb:114]: Curious about this change, why is the location at a different level for plugins?

File: gem2rpm/foreman_plugin.spec.erb [hunk 2]
```diff
@@ -77,6 +76,13 @@ BuildRequires: gcc
 BuildArch: noarch
 <% end -%>
 Provides: foreman-plugin-%{plugin_name} = %{version}
+<% if has_assets || has_webpack -%>
+<% for d in spec.runtime_dependencies -%>
+<% for req in d.requirement -%>
+BuildRequires: rubygem(<%= d.name %>)<%= " #{req}" unless req.empty? %>
+<% end -%>
+<% end -%>
+<% end -%>
```
Comment (ehelms) [inline gem2rpm/foreman_plugin.spec.erb:85]: Sad :/ What step of the build related to assets is it that ends up requiring all the runtime dependencies present? (my memory fails me)

File: gem2rpm/foreman_plugin.spec.erb (hunk 1 above)
Comment (ekohl) [reply gem2rpm/foreman_plugin.spec.erb:114]: This is aligning it to `default.spec.erb` where we made the same change.

File: gem2rpm/foreman_plugin.spec.erb (hunk 2 above)
Comment (ekohl) [reply gem2rpm/foreman_plugin.spec.erb:85]: `foreman-rake assets:precompile[$plugin]`

File: gem2rpm/foreman_plugin.spec.erb (hunk 1 above)
Comment (ehelms) [reply gem2rpm/foreman_plugin.spec.erb:114]: Oh right, the `../` threw me.

**Review by ehelms — APPROVED** [review]

File: gem2rpm/default.spec.erb [hunk 3]
```diff
@@ -6,6 +6,7 @@ LICENSES = JSON.load(File.read(File.join(root, 'licenses.json')))
 
 config.rules[:doc] << /\/?CHANGES/
 config.rules[:doc] << 'sample'
+config.rules[:doc] << 'example'
 config.rules[:doc] << /\/?SECURITY/
```
Comment (ekohl) [inline gem2rpm/default.spec.erb:10]: So I found out why this didn't appear to work anymore: https://github.com/fedora-ruby/gem2rpm/issues/115

**Comment by ehelms** [comment]
I'd be OK taking in these round of changes as it will unblock a set of packages to get built.

**Pull request merged**